integration-tests/src/test/java/tachyon/master/lineage/LineageMasterIntegrationTest.java

```diff
@@ -152,6 +164,39 @@ public void lineageCompleteAndAsyncPersistTest() throws Exception {
     }
   }
 
+  /**
+   * Tests that a lineage job is executed when the output file for the lineage is reported as lost.
+   */
+  @Test(timeout = 20000)
+  public void lineageRecoveryTest() throws Exception {
+    File logFile = mFolder.newFile();
+    logFile.delete();
```

Comment:
we need it so that we can wait on `logFile.exists()` below. I've added comments to make this less confusing.